test/utils/helpers.ts

```diff
@@ -201,4 +211,31 @@ export const bumpFeesForBtcTx = async (
     bitcoinCoreClient: BitcoinCoreClient,
     txId: string, 
     options: RbfOptions = {conf_target: 72, estimate_mode: RbfEstimateMode.economical}
-): Promise<RbfResponse> => bitcoinCoreClient.client.command("bumpfee", txId, options);
\ No newline at end of file
+): Promise<RbfResponse> => bitcoinCoreClient.client.command("bumpfee", txId, options);
+
+/**
+ * Wait for an event to show up as finalized.
+ * 
+ * This method will only subscribe to new blocks once called and then look for a specific event. 
+ * Contrary to @see {@link DefaultTransactionAPI.waitForEvent} which looks at all past events 
+ * and may return due to older events using the same name.
+ * 
+ * Note: This method checks the parent blocks in case the event we are looking for 
+ * was finalized in a block just before this method was called.
+ * 
+ * @param interBtcApi the api to use for querying
+ * @param event the event to wait for, eg. `api.events.assetRegistry.RegisteredAsset`
+ */
+export const waitForFinalizedEvent = async <T extends AnyTuple>(
```

Comment:
Now that this is refactored, it can replace the duplicated code [here](https://github.com/interlay/interbtc-api/blob/d8446ca0eb272af29212ec646df4531d1e7800da/test/integration/parachain/release/redeem.test.ts#L138) and [here](https://github.com/interlay/interbtc-api/blob/d8446ca0eb272af29212ec646df4531d1e7800da/test/integration/parachain/staging/sequential/replace.test.ts#L102)

Reply:
Yep, that's part of my clean up / follow up task for the PR that introduced those bits.